tracer/src/Datadog.Trace/DuckTyping/DuckType.Methods.cs

```diff
@@ -687,7 +687,7 @@ private static class MethodIlHelper
                 MethodInfo outerMethod,
                 ParameterInfo[] outerMethodParameters,
                 Type[] outerMethodGenericArguments,
-                Func<LazyILGenerator, Type, Type> duckCastParameterFunc,
+                Func<LazyILGenerator, Type, Type, Type> duckCastParameterFunc,
```

Comment:
I think it's getting complex enough that we should introduce a custom delegate so we can name the arguments (for another PR)